ocs_ci/ocs/scale_pgsql.py

```diff
@@ -0,0 +1,179 @@
+from ocs_ci.ocs import constants, machine, ocp
+from subprocess import CalledProcessError
+from ocs_ci.ocs import machine as machine_utils
+from ocs_ci.framework import config
+
+log = logging.getLogger(__name__)
+
+
+class ScalePodPGSQL(Postgresql):
+    """
+    Scale Postgresql workload with scale parameters and functions
+    """
+
+    def __init__(self, node_selector=constants.SCALE_NODE_SELECTOR, **kwargs):
+        """
+        Initializer function
+
+        """
+        super().__init__(**kwargs)
+        self._node_selector = node_selector
+
+    def apply_crd(self, crd):
+        """
+        Apply the CRD
+
+        Args:
+            crd (str): yaml to apply
+
+        """
+        Postgresql.apply_crd(self, crd=crd)
+
+    def setup_postgresql(self, replicas, node_selector=None):
+        log.info("Deploying postgres database")
+        try:
+            pgsql_service = templating.load_yaml(
+                constants.PGSQL_SERVICE_YAML
+            )
+            pgsql_cmap = templating.load_yaml(
+                constants.PGSQL_CONFIGMAP_YAML
+            )
+            pgsql_sset = templating.load_yaml(
+                constants.PGSQL_STATEFULSET_YAML
+            )
+            pgsql_sset['spec']['replicas'] = replicas
+            if node_selector is not None:
+                pgsql_sset['spec']['template']['spec'][
+                    'nodeSelector'] = node_selector
+            self.pgsql_service = OCS(**pgsql_service)
+            self.pgsql_service.create()
+            self.pgsql_cmap = OCS(**pgsql_cmap)
+            self.pgsql_cmap.create()
+            self.pgsql_sset = OCS(**pgsql_sset)
+            self.pgsql_sset.create()
+            self.pod_obj.wait_for_resource(
+                condition='Running',
+                selector='app=postgres',
+                resource_count=replicas,
+                timeout=3600
+            )
+        except (CommandFailed, CalledProcessError) as cf:
+            log.error('Failed during setup of PostgreSQL server')
+            raise cf
+        self.pgsql_is_setup = True
+        log.info("Successfully deployed postgres database")
+
+    def _create_pgbench_benchmark(
+        self, replicas, clients=None, threads=None,
+        transactions=None, scaling_factor=None,
+        timeout=None
+    ):
+        Postgresql.create_pgbench_benchmark(
+            self, replicas, clients=clients,
+            threads=threads, transactions=transactions,
+            scaling_factor=scaling_factor, timeout=timeout
+        )
+
+    def _wait_for_pgbench_status(self, status, timeout=None):
+        Postgresql.wait_for_postgres_status(
+            self, status=status, timeout=timeout
+        )
+
+    def _get_pgbench_pods(self):
+        Postgresql.get_pgbench_pods()
+
+    def _validate_pgbench_run(self, pgbench_pods, print_table=True):
+        Postgresql.validate_pgbench_run(self, pgbench_pods)
+
+    def add_new_node_and_label_with_scale_label(self, machineset_name):
```

Comment:
@tiffanyn108 as discussed please refer function  check_and_add_enough_worker() in scale_lib.py, this will also help in testing the code for other platforms too.
https://github.com/red-hat-storage/ocs-ci/blob/de691f3689b79606a0e8a726fa4d47d10250d232/ocs_ci/ocs/scale_lib.py#L541

Reply:
Acked.  Updated add_worker_node function using the same concept with scale_lib.